monkey/infection_monkey/island_api_client/http_island_api_client.py

```diff
@@ -44,34 +44,58 @@ class HTTPIslandAPIClient(IIslandAPIClient):
     A client for the Island's HTTP API
     """
 
+    TOKEN_HEADER_KEY = "Authentication-Token"
+
     def __init__(
-        self, agent_event_serializer_registry: AgentEventSerializerRegistry, http_client: HTTPClient
+        self,
+        agent_event_serializer_registry: AgentEventSerializerRegistry,
+        http_client: HTTPClient,
+        otp: str,
     ):
         self._agent_event_serializer_registry = agent_event_serializer_registry
-        self.http_client = http_client
+        self._http_client = http_client
+        self._otp = otp
 
     def connect(
         self,
         island_server: SocketAddress,
     ):
-        self.http_client.connect(island_server)
+        try:
+            self._http_client.server_url = f"{island_server}/api/"
+            self._http_client.get("", params={"action": "is-up"})
+        except Exception as err:
+            logger.debug(f"Connection to {island_server} failed: {err}")
+            self._http_client.server_url = None
+            raise err
+
+        try:
+            auth_token = self._get_authentication_token()
+        except Exception as err:
+            logger.error("Agent authentication failed")
+            raise err
+
+        self._http_client.additional_headers = {HTTPIslandAPIClient.TOKEN_HEADER_KEY: auth_token}
+
+    def _get_authentication_token(self) -> str:
+        response = self._http_client.post("/agent-otp-login", {"otp": self._otp})
+        return response.json()["token"]
 
     def get_agent_binary(self, operating_system: OperatingSystem) -> bytes:
         os_name = operating_system.value
-        response = self.http_client.get(f"agent-binaries/{os_name}")
+        response = self._http_client.get(f"/agent-binaries/{os_name}")
```

Comment:
Why add "/" to all of these and then strip it out in `HTTPClient._send_request()`?

Reply:
Wheather there's a `/` prepended or not doesn't matter. Because of that I didn't change these back